airbyte-integrations/bases/debezium/src/main/java/io/airbyte/integrations/debezium/internals/mysql/MySqlDebeziumStateUtil.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.debezium.internals.mysql;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.annotations.VisibleForTesting;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.db.jdbc.JdbcDatabase;
+import io.airbyte.db.jdbc.JdbcUtils;
+import io.airbyte.integrations.debezium.internals.AirbyteFileOffsetBackingStore;
+import io.airbyte.integrations.debezium.internals.AirbyteSchemaHistoryStorage;
+import io.airbyte.integrations.debezium.internals.DebeziumRecordPublisher;
+import io.airbyte.protocol.models.v0.ConfiguredAirbyteCatalog;
+import io.debezium.engine.ChangeEvent;
+import java.sql.SQLException;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MySqlDebeziumStateUtil {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(MySqlDebeziumStateUtil.class);
+  public static final String MYSQL_CDC_OFFSET = "mysql_cdc_offset";
+  public static final String MYSQL_DB_HISTORY = "mysql_db_history";
+
+  public JsonNode constructInitialDebeziumState(final Properties properties,
```

Comment:
So this method is basically constructing the state from spinning up the DBZ engine and grabbing the state assc with the first record?

Reply:
Yes and no! As discussed over call, we construct the offset by ourselves but in order to construct the db schema we need Debezium and thus we give Debezium the offset that we generated and run it with `schema_only_recovery` mode and it generates the schema history on its own and we use it.